src/mocks.ts

```diff
@@ -40,19 +40,29 @@ export const mockManifest = () => {
     'client/index.html': `client.${HASH}`,
   })
 }
-let cacheStore: any = {}
+let cacheStore: any = new Map()
 export const mockCaches = () => {
   return {
     default: {
-      match: (key: Request) => {
-        const url = key.url
-        return cacheStore[url] || null
+      match: (key: any) => {
+        let cacheKey = {
+          url: key.url,
+          headers: {
+            'etag': `${key.headers.has('if-none-match') ? key.headers.get('if-none-match')  : null}`
```

Comment:
If `key.headers` is a spec-compliant Headers object, then I would expect this to be equivalent:

```suggestion
            'etag': key.headers.get('if-none-match')
```

because Headers.get() already returns `null` if the header doesn't exist. (That said, since I don't think this is CF Workers code, I may be wrong.)

Reply:
Good catch here Harris -- I was following [the MDN reference](https://developer.mozilla.org/en-US/docs/Web/API/Headers/get#Parameters) which is either confusing or wrong. I can't find a scenario where a `TypeError` is returned on `Headers.get()`, so will refactor this

This is indeed not Workers code. @victoriabernard92 is using `service-worker-mock` which is the de facto standard library for Service Worker mocks, so this is a stub (in progress 😬) for the CF implementation of the Cache API interface.

Does Cache API rely on CDN cache to ignore headers? Or are request headers deleted the runtime, before writing to/querying cache?